examples/with-apollo/lib/withData.js

```diff
@@ -0,0 +1,49 @@
+import { ApolloProvider, getDataFromTree } from 'react-apollo'
+import React from 'react'
+import 'isomorphic-fetch'
+import { initClient } from './initClient'
+import { initStore } from './initStore'
+
+export default (Component) => (
+  class extends React.Component {
+    static async getInitialProps (ctx) {
+      const headers = ctx.req ? ctx.req.headers : {}
+      const client = initClient(headers)
+      const store = initStore(client, client.initialState)
+
+      if (!process.browser) {
+        const app = (
+          <ApolloProvider client={client} store={store}>
+            <Component url={{ query: ctx.query, pathname: ctx.pathname }} />
```

Comment:
`Component` potentially has a `getInitialProps`, I think it should be conditionally called and its result injected in here and also returned below at line 24. 

As I did in my app: https://github.com/relatenow/relate/blob/77927a28adb5a8b22071cce6040c21917a9178ae/hocs/withData.js#L32